src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -531,8 +531,6 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
             }
             catch (Exception e)
             {
-                ClusterStateCache.UpdateClusterState(Settings.Host!, Settings.Port, ClusterState.Offline,
```

Comment:
Should we differentiate between PostgresException (e.g. unknown user) and other exceptions (typically networking), as discussed offline? We could even look at PostgresException.IsTransient - a transient exception is typically means that the cluster is down, etc.

Reply:
But that's exactly the thing it does - `Break` marks the cluster as down only when the exception is `NpgsqlException` and the inner exception is `IOException`.